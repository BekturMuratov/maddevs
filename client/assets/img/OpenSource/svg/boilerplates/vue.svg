<svg xmlns="http://www.w3.org/2000/svg" width="336" height="125" fill="none" xmlns:v="https://vecta.io/nano"><g clip-path="url(#A)"><path d="M206.784.001l-38.797 67.203L129.191.001H-.008l167.996 290.997L335.984.001h-129.2z" fill="#41b883"/><path d="M206.785.001l-38.797 67.203L129.191.001h-62l100.798 174.595L268.786.001h-62.001z" fill="#34495e"/></g><defs><clipPath id="A"><path fill="#fff" d="M0 0h336v291H0z"/></clipPath></defs></svg>