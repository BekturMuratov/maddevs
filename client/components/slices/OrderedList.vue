<template>
  <ListNumberedBox>
    <PrismicRichText
      v-if="hasListIntroduction"
      :field="slice.primary.list_introduction"
      :html-serializer="htmlSerializer"
    />
    <ListNumberedItemBox
      v-for="(item, index) in slice.items"
      :key="index"
    >
      {{ $prismic.asText(item.list_item) }}
    </ListNumberedItemBox>
  </ListNumberedBox>
</template>

<script>
import ListNumberedBox from '@/components/Cases/shared/ListNumberedBox'
import ListNumberedItemBox from '@/components/Cases/shared/ListNumberedItemBox'

export default {
  name: 'OrderedList',
  components: {
    ListNumberedItemBox,
    ListNumberedBox,
  },

  props: {
    slice: {
      type: Object,
      required: true,
    },

    htmlSerializer: {
      type: Function,
      default: () => null,
    },
  },

  computed: {
    hasListIntroduction() {
      return this.slice.primary.list_introduction && this.slice.primary.list_introduction.length !== 0
    },
  },
}
</script>

<style lang="scss" scoped>
.case-list_numbered {
  margin: 35px 0;
}
</style>
