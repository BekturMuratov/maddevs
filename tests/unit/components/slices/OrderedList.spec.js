import { render } from '@testing-library/vue'
import { shallowMount } from '@vue/test-utils'
import OrderedList from '@/components/slices/OrderedList'

const slice = {
  items: [
    { list_item: 'list item 1' },
    { list_item: 'list item 2' },
    { list_item: 'list item 3' },
    { list_item: 'list item 4' },
    { list_item: 'list item 5' },
  ],
  primary: {
    list_introduction: 'list introduction',
  },
}
const stubs = ['PrismicRichText']

const mocks = {
  $prismic: {
    asText: text => text,
    asHtml: text => text,
  },
}

describe('orderedListSlice component', () => {
  it('should render correctly with data', () => {
    const { container } = render(OrderedList, {
      props: {
        slice,
        htmlSerializer: () => null,
      },
      stubs,
      mocks,
    })
    expect(container).toMatchSnapshot()
  })
  it('the htmlSerializer should to be return null', () => {
    const wrapper = shallowMount(OrderedList, {
      propsData: {
        slice: {
          items: [],
          primary: {
            list_introduction: '',
          },
        },
        htmlSerializer: () => null,
      },
      stubs: ['PrismicRichText'],
      mocks: {},
    })
    expect(wrapper.vm.htmlSerializer()).toBeNull()
  })
})
