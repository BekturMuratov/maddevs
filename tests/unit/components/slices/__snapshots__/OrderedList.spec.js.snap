// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`orderedListSlice component should render correctly with data 1`] = `
<div>
  <ol
    class="case-list_numbered"
  >
    <prismicrichtext-stub
      field="list introduction"
      html-serializer="function htmlSerializer() {
          return null;
        }"
    />
     
    <li
      class="case_list-numbered_item"
    >
      
    list item 1
  
    </li>
    <li
      class="case_list-numbered_item"
    >
      
    list item 2
  
    </li>
    <li
      class="case_list-numbered_item"
    >
      
    list item 3
  
    </li>
    <li
      class="case_list-numbered_item"
    >
      
    list item 4
  
    </li>
    <li
      class="case_list-numbered_item"
    >
      
    list item 5
  
    </li>
  </ol>
</div>
`;
